richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -132,7 +132,8 @@ public String substring(int from) {
             int segIdx = pos.getMajor();
             List<SEG> segs = new ArrayList<>(segIdx + 1);
             segs.addAll(segments.subList(0, segIdx));
-            segs.add(segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor()));
+            segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
+                            .ifPresent(seg -> segs.add(seg));
```

Comment:
I'm afraid that this has the potential of breaking the invariant that [each Paragraph has at least one segment](https://github.com/afester/RichTextFX/blob/e40cf78eba2c3adcca77c52554a3fa2e26ee6c94/richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java#L42).